test/ziggurat/producer_test.clj

```diff
@@ -0,0 +1,27 @@
+(ns ziggurat.producer-test
+  (:require [clojure.test :refer :all]
+            [flatland.protobuf.core :as proto]
+            [ziggurat.streams :refer [start-streams stop-streams]]
+            [ziggurat.fixtures :as fix]
+            [ziggurat.config :refer [ziggurat-config]]
+            [ziggurat.producer :as producer]))
+
+(use-fixtures :once fix/mount-only-config-and-producer)
+
+(deftest properties-map-is-generated-correctly
+  (is (= "localhost:9092" (.get (producer/producer-config-properties) "bootstrap.servers")))
+  (is (= "all" (.get (producer/producer-config-properties) "acks")))
+  (is (= 5 (.get (producer/producer-config-properties) "retries")))
+  (is (= 5 (.get (producer/producer-config-properties) "max.in.flight.requests.per.connection")))
+  (is (= "org.apache.kafka.common.serialization.StringSerializer" (.get (producer/producer-config-properties) "key.serializer")))
+  (is (= "org.apache.kafka.common.serialization.StringSerializer" (.get (producer/producer-config-properties) "value.serializer")))
+  (is (false? (.get (producer/producer-config-properties) "enable.idempotence"))))
+
+(deftest send-data-with-topic-and-value-test
+  (let [topic "test-topic"
+        data "Hello World!!!"]
+    (let [future (producer/send topic data)]
```

Comment:
can this be in the last `let` only? like:
```clj
    (let [  topic "test-topic"
              data "Hello World!!!"
              future (producer/send topic data)]
```
also, what happens if the `topic` is nil? 🤔

Reply:
@riteeks You're right. But, I think these two lets logically divide the code in two parts:

1. First let setups the data for calling send()
2. Second let simply acts on the result of calling send()

Mixing these two, IMHO, will lead to a less readable code.

Reply:
> Mixing these two, IMHO, will lead to a less readable code.

make sense. 

> send in ziggurat is just a wrapper around KafkaProducer#send. Thus, whatever arguments are passed will just be passed "as they are" to KafkaProducer. That class will be responsible for error handling and exceptions. Ziggurat's code is just a thin layer in-front
of KafkaProducer.

for this, I was just thinking of having a test for bad-path also for the ziggurat's `send` function. But I am not very sure how much value will this add.